river/compose/select.py

```diff
@@ -105,6 +136,12 @@ def __init__(self, *keys: typing.Tuple[base.typing.FeatureName]):
     def transform_one(self, x):
         return {i: x[i] for i in self.keys}
 
+    def transform_many(self, X: pd.DataFrame):
+        # INFO: has either side-effects or doesn't have copy - choose your poison
+        # REFLECTION: worth adding `copy=True` parameter to the object constructor to allow both?
+        return X.loc[:, self.keys].copy()
```

Comment:
The expectation is that every `transform_` method in River is pure (with no side-effect). I would default to that behavior. We can think about performance later :)

Reply:
Fair :)